cms/datasets/models.py

```diff
@@ -15,16 +17,66 @@
 
 # TODO: This needs a revisit. Consider caching + fewer requests
 class ONSDatasetApiQuerySet(APIQuerySet):
+    def __init__(self, *args: object, **kwargs: object) -> None:
+        super().__init__(*args, **kwargs)
+        self.token: str | None = None
+
+    def with_token(self, token: str) -> "ONSDatasetApiQuerySet":
+        """Return a cloned queryset with the given authentication token.
+
+        We clone the queryset to ensure the method is stateless and doesn't mutate
+        the original queryset. This allows chaining operations (e.g.,
+        ONSDataset.objects.with_token(token).filter(...).all()) without affecting
+        other code that may be using the same base queryset instance.
+
+        Args:
+            token: Bearer token for API authentication
+
+        Returns:
+            Cloned queryset with token attached
+        """
+        clone: ONSDatasetApiQuerySet = self.clone()
+        clone.token = token
+        return clone
+
     def get_results_from_response(self, response: Mapping) -> Iterable:
         results: Iterable = response["items"]
         return results
 
     def fetch_api_response(self, url: str | None = None, params: Mapping | None = None) -> dict:
-        api_response: dict = super().fetch_api_response(url=url, params=params)
+        # Add Authorization header if token is set
+        headers = dict(self.http_headers) if self.http_headers else {}
+        if self.token:
+            headers["Authorization"] = f"Bearer {self.token}"
+
+        if url is None:
+            url = self.base_url
+        if params is None:
+            params = {}
+
+        try:
+            response = requests.get(url, params=params, headers=headers, timeout=30)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                logger.warning("Rate limit exceeded when fetching datasets: %s", url)
+                raise
+            # Check for 5xx server errors (500-599)
+            if e.response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                logger.error("Server error when fetching datasets: %s - Status: %s", url, e.response.status_code)
+                raise
+            raise
+        except requests.exceptions.RequestException as e:
+            logger.error("Request failed when fetching datasets: %s - Error: %s", url, e)
```

Comment:
Suggestion: Use `logger.exception` with `extra` param for `url`.

Reply:
Added `extra` in [0308492](https://github.com/ONSdigital/dis-wagtail/pull/389/commits/0308492c0c4a00cca864bdc13b5ab10333a9f79e)